zerver/actions/message_send.py

```diff
@@ -535,35 +542,54 @@ def maybe_add_event(user_profile_id: int, bot_type: int) -> None:
             return
 
         is_stream = recipient_type == Recipient.STREAM
+        bot_service = get_bot_services(user_profile_id)
```

Comment:
I'm guessing we fetch that in this code path somewhere else; it would be best to not have to do it twice.